mail/urls.py

```diff
@@ -16,4 +16,9 @@
     url(r'^api/v0/mail/course/(?P<course_id>[\d]+)/$', CourseTeamMailView.as_view(), name='course_team_mail_api'),
     url(r'^api/v0/mail/learner/(?P<student_id>[\d]+)/$', LearnerMailView.as_view(), name='learner_mail_api'),
     url(r'^api/v0/mail/automatic_email/$', AutomaticEmailView.as_view(), name='automatic_email_api'),
+    url(
```

Comment:
Having two urls named the same could cause some issues if they're used with `reverse()` at some point. I also think you can update this so that you only need one definition. I tested these changes locally so it should be a simple copy/paste:

```python
from django.conf.urls import url, include  # added include
from rest_framework import routers

router = routers.DefaultRouter()
router.register(r'automatic_email', AutomaticEmailView, base_name='automatic_email_api')

urlpatterns = [
    ...
    url(r'^api/v0/mail/', include(router.urls)),  # replaces lines 18-22
]
```

Then add this to `AutomaticEmailView`:
```python
from rest_framework.viewsets import GenericViewSet  # add import

class ProfileViewSet(RetrieveModelMixin, UpdateModelMixin, GenericViewSet):  # add GenericViewSet
    ...
    lookup_value_regex = '[-\w.]+'  # pylint: disable=anomalous-backslash-in-string

```